File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.java
import java.util.PriorityQueue;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        PriorityQueue<Integer> pq = new PriorityQueue();
        int result = 0;

        for (int i = 0; i < n; i++) {
            pq.add(input.nextInt());
        }

        while (pq.size() > 1) {
            int a = pq.poll();
            int b = pq.poll();
            result += a + b;
            pq.add(a + b);
        }

        System.out.println(result);
    }
}

File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.kt
import java.util.*

fun main() {
    val n = readLine()!!.toInt()
    val cards = PriorityQueue<Int>()
    var result = 0

    repeat(n) {
        cards.add(readLine()!!.toInt())
    }

    while (cards.size > 1) {
        val a = cards.poll()
        val b = cards.poll()
        result += a + b
        cards.add(a + b)
    }

    print(result)
}